core/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDAndVersionSeqNoLookup.java

```diff
@@ -113,4 +121,21 @@ private int getDocID(BytesRef id, Bits liveDocs) throws IOException {
             return DocIdSetIterator.NO_MORE_DOCS;
         }
     }
+
+    /** Return null if id is not found. */
+    public DocIdAndSeqNo lookupSequenceNo(BytesRef id, Bits liveDocs, LeafReaderContext context) throws IOException {
```

Comment:
I think that this method can be package private? And maybe the other lookup methods in this class too?